src/main/java/fi/aalto/cs/apluscourses/intellij/toolwindows/APlusToolWindowFactory.java

```diff
@@ -85,6 +86,8 @@ private static ExercisesView createExercisesView(@NotNull Project project) {
     MainViewModel mainViewModel = PluginSettings.getInstance().getMainViewModel(project);
 
     ExercisesView exercisesView = new ExercisesView();
+    mainViewModel.courseViewModel.addValueObserver(exercisesView, ((observer, value) -> observer.setCourseName(
+        Optional.ofNullable(value).map(c -> c.getModel().getName()).orElse(""))));
```

Comment:
The view shouldn't have setter, instead it should get notified of a change and update itself appropriately. That way, we can modify the view independently of the rest of the system. The logic here (and in `ExercisesView::setCourseName`), should ideally be moved to the view model so that the view could stay as "dumb" as possible.